eth/plasma.go

```diff
@@ -70,42 +67,67 @@ func InitPlasma(contractAddr common.Address, client Client, finalityBound uint64
 		client:          client,
 		logger:          logger,
 
-		ethBlockNum:   ethBlockNum,
-		blockNum:      lastCommittedBlock,
+		commitmentRate:      commitmentRate,
+		lastBlockSubmission: time.Now(),
+
 		finalityBound: finalityBound,
 
 		Mutex: &sync.Mutex{},
 	}
 
-	// listen to new ethereum block headers
-	ethCh, err := client.SubscribeToHeads()
-	if err != nil {
-		return nil, fmt.Errorf("could not successfully subscribe to heads: { %s }", err)
-	}
-
-	go watchEthBlocks(plasma, ethCh)
-
 	return plasma, nil
 }
 
 func (plasma *Plasma) OperatorAddress() (common.Address, error) {
 	return plasma.contract.Operator(nil)
 }
 
-// SubmitBlock proxy. TODO: handle batching with a timmer interrupt
-func (plasma *Plasma) SubmitBlock(block plasmaTypes.Block) error {
-	// only the contract operator can submit blocks
-	if plasma.operatorSession == nil {
+// SubmitBlock proxy. Blocks are committed ever > commitment interval
+func (plasma *Plasma) CommitPlasmaHeaders(ctx sdk.Context, plasmaStore store.PlasmaStore) error {
+	// only the contract operator can submit blocks. The commitment duration must also pass
+	if plasma.operatorSession == nil || time.Since(plasma.lastBlockSubmission).Seconds() < plasma.commitmentRate.Seconds() {
 		return nil
 	}
 
-	plasma.blockNum = plasma.blockNum.Add(plasma.blockNum, big.NewInt(1))
+	plasma.logger.Info("attempting to commit plasma headers...")
+
+	lastCommittedBlock, err := plasma.contract.LastCommittedBlock(nil)
+	if err != nil {
+		plasma.logger.Error("error retrieving the last committed block number")
+		return err
+	}
 
-	_, err := plasma.operatorSession.SubmitBlock(
-		[][32]byte{block.Header},
-		[]*big.Int{big.NewInt(int64(block.TxnCount))},
-		[]*big.Int{block.FeeAmount},
-		plasma.blockNum)
+	firstBlockNum := new(big.Int).Add(lastCommittedBlock, utils.Big1)
+	blockNum := lastCommittedBlock.Add(lastCommittedBlock, utils.Big1)
```

Comment:
Why use lastCommittedBlock and then make blockNum? Couldn't this all work with just blockNum or blockNum/firstBlockNum?

Reply:
blockNum is incremented while grabbing blocks.

I could subtract the number of  headers from the final block number but that's the same as just holding a reference to the first block numbers.

`blockNum` is actually not a new `big.Int`, its the same reference to `lastCommittedBlock`. `firstBlockNum` creates a new one though.